R/db.R

```diff
@@ -284,11 +284,11 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns, blacklist)
     if (length(blacklist) > 0L) {
       df_without_id <- df[-which(names(df) == key)]  # Don't blacklist on id.
       if (any(is.na(blacklist))) {
-        df <- df[apply(!is.na(df_without_id), 1, all),]
+        df <- df[apply(!is.na(df_without_id), 1, all), , drop = FALSE]
         blacklist <- blacklist[!is.na(blacklist)]
       }
       if (length(blacklist) > 0L) {
-        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))),]
+        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))), , drop = FALSE]
```

Comment:
@khalidharun Agreed that we need a faster `apply(., 1, ...)`. Looking through plyr/dplyr doesn't yield anything at the moment -- the best I've found is http://stackoverflow.com/questions/13978918/apply-is-slow-how-to-make-it-faster-or-what-are-my-alternatives but it doesn't give us what we want.